tests/execute/restraint/report_result/test.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "tmp=\$(mktemp -d)" 0 "Create tmp directory"
+        rlRun "set -o pipefail"
+        rlRun "pushd data"
+    rlPhaseEnd
+
+    rlPhaseStartTest 'Verify mocked Restraint rstrnt-report-result file generated correctly.'
+        rlRun "tmt run -vvvddd --remove"
```

Comment:
Here we can extend coverage with additional checks that good/bad test was reported correctly.

Reply:
Extended with additional tests.